client/src/main/webjar/authentication/authentication.config.js

```diff
@@ -0,0 +1,62 @@
+(function (angular) {
+    "use strict";
+
+    var module = angular.module("healthBam.authentication");
+
+    /**
+     * Returns the protocol, host, and port part of our single-page-app's URL.
+     *
+     * Right now, this has to match what the server is sending to google, so it
+     * has to be explicit with the port.
+     */
+    function getUrlPrefix() {
+        /* location.protocol includes the colon */
+        var urlPrefix = location.protocol + "//" + location.hostname,
+            port = location.port;
+
+        if (!port) {
+            /* we have to infer it from the protocol */
+            if (location.protocol === "http:") {
+                port = 80;
+            } else if (location.protocol === "https:") {
+                port = 443;
+            } else {
+                throw new Error("Unknown protocol! " + location.protocol);
+            }
+        }
+
+        return urlPrefix + ":" + port + "/";
+    }
+
+    /**
+     * Configures satellizer's Google Oauth.
+     */
+    function configGoogleOauth(
+        $authProvider
+    ) {
+        var redirectUri = getUrlPrefix() + "views/oauth-callback";
+
+        window.console.log("Registering google oauth with: redirectUri = ", redirectUri);
+
+        $authProvider.google(
+            {
+                /* https://developers.google.com/identity/protocols/OAuth2UserAgent */
+                prompt: "select_account",
+                optionalUrlParams: [
+                    "prompt"
+                ],
+                url: "/auth/google", /* posts to our server */
```

Comment:
I'm not sure if a relative path would work too, but the path we are posting to is https://hmhb-ryan.herokuapp.com/auth/google.  It's possible our base tag comes into play, however, it wouldn't surprise me if the new window could screw this up (so I was explicit on it).  I can try it out if you think it would be better.